core/src/main/resources/xtdb/sql/parser/sql.ebnf

```diff
@@ -384,6 +384,12 @@ correlation_name
     | boolean_type
     | datetime_type
     | interval_type
+    | duration_type
+    | varchar_type
+    ;
+
+varchar_type
```

Comment:
Separate VARCHAR type - open to think a bit more about this, but `character_string_type`/`character_large_object_type` currently isnt being used anywhere, and it made sense to create a separate "VARCHAR" type than use `character_string_type`.

In any case - I dont think we want to use the current VARCHAR type in `character_string_type` as a cast target, otherwise casting to a string/varchar would look like this:
- `CAST(DATE '2011-01-01' AS VARCHAR(n))`
(Which would require a user to specify a character length)

Reply:
I think `VARCHAR(n)` is the correct way to go here, its both spec compliant and standard behaviour in other databases. 

If we wish to add a no-limit version as well (e.g. `VARCHAR`) I'm not against that.